Mage.Sets/src/mage/cards/k/KaitoDancingShadow.java

```diff
@@ -129,17 +82,15 @@ public KaitoDancingShadowEffect copy() {
 
     @Override
     public boolean apply(Game game, Ability source) {
-        KaitoDancingShadowWatcher watcher = game.getState().getWatcher(KaitoDancingShadowWatcher.class);
-        if (watcher == null) {
-            return false;
-        }
-        TargetCreatureThatDealtCombatDamage target = new TargetCreatureThatDealtCombatDamage(0, 1, watcher.getPermanents());
+        FilterPermanent filter = new FilterControlledCreaturePermanent();
+        filter.add(new DamagedPlayerThisTurnPredicate(TargetController.ANY));
```

Comment:
So ideally, this effect would take the set of creatures from the trigger, to work correctly in edge cases with multiple combat phases. If that's too annoying to make work, then at least use the two-parameter constructor for this predicate with combatDamageOnly  = true.